ndcube/utils/wcs.py

```diff
@@ -300,3 +152,257 @@ def _pixel_keep(wcs_object):
+    ----------
+    axis: `int` of `numpy.ndarray` of `int`
+        The axis number(s) before reflection.
+
+    naxes: `int`
+        The number of array axes.
+
+    Returns
+    -------
+    reflected_axis: `int` of `numpy.ndarray` of `int
+        The axis number(s) after reflection.
+    """
+    if axis is None:
+        return None
+    # Check type of input.
+    if isinstance(axis, (numbers.Integral, np.ndarray)):
+        raise TypeError(f"axis must be int or array of ints. Got: {type(axis)}.")
+    # Convert scalar input to array so algorithm can be vectorized.
+    if np.isscalar(axis):
+        input_is_scalar = True
+        axis = np.array([axis])
+    else:
+        input_is_scalar = False
+        # Check input array is of int type.
+        if not isinstance(axis.dtype, numbers.Integral):
+            raise TypeError(f"if axis is array, dtype must be int.  Got: {axis.dtype}")
+    # Convert negative indices to positive equivalents.
+    idx_neg = axis < 0
+    axis[idx_neg] = np.abs(axis[idx_neg]) - 1
+    if any(axis > naxes - 1):
+        raise IndexError("Axis out of range.  "
+                         f"Number of axes = {naxes}; Axis numbers requested = {axes}"))
+    # Reflect axis about center of number of axes.
+    reflected_axis = naxes - 1 - axis
+    # Ensure output is returned as scalar if input was scalar.
+    if input_is_scalar:
+        reflected_axis = reflected_axis[0]
+
+    return reflected_axis
+
+
+def pixel_axis_to_world_axes(pixel_axis, axis_correlation_matrix):
+    """
+    Retrieves the indices of the world axis physical types corresponding to a pixel axis.
+
+    Parameters
+    ----------
+    pixel_axis: `int`
+        The pixel axis index/indices for which the world axes are desired.
+
+    axis_correlation_matrix: `numpy.ndarray` of `bool`
+        2D boolean correlation matrix defining the dependence between the pixel and world axes.
+        Format same as `astropy.wcs.BaseLowLevelWCS.axis_correlation_matrix.
+
+    Returns
+    -------
+    world_axes: `numpy.ndarray`
+        The world axis indices corresponding to the pixel axis.
+    """
+    return np.arange(axis_correlation_matrix.shape[0])[axis_correlation_matrix[:, pixel_axis]]
+
+
+def world_axis_to_pixel_axes(world_axis, axis_correlation_matrix):
+    """
+    Gets the pixel axis indices corresponding to the index of a world axis physical type.
+
+    Parameters
+    ----------
+    world_axis: `int`
+        The index of the physical type for which the pixes axes are desired.
+
+    axis_correlation_matrix: `numpy.ndarray` of `bool`
+        2D boolean correlation matrix defining the dependence between the pixel and world axes.
+        Format same as `astropy.wcs.BaseLowLevelWCS.axis_correlation_matrix.
+
+    Returns
+    -------
+    pixel_axes: `numpy.ndarray`
+        The pixel axis indices corresponding to the world axis.
+    """
+    return np.arange(axis_correlation_matrix.shape[1])[axis_correlation_matrix[world_axis]]
```

Comment:
why not make this one function with an option to select direction, that's a lot of LOC you are saving :wink:

Reply:
Doesn't that go against what you said about my functions allowing too many options? ;P  And I would actually have to add a couple lines of code and make this API a little more confusing for the sake of saving on a docstring.